cborg/src/Codec/CBOR/Read.hs

```diff
@@ -1163,6 +1213,42 @@ tryConsumeBytes hdr !bs = case fromIntegral hdr :: Word of
   _    -> DecodeFailure
 
 
+{-# INLINE tryConsumeByteArray #-}
+tryConsumeByteArray :: Word8 -> ByteString -> DecodedToken (LongToken BA.ByteArray)
+tryConsumeByteArray hdr !bs = case fromIntegral hdr :: Word of
```

Comment:
Surely this is the wrong way to factor things? Don't we just use the normal `tryConsumeBytes` and put the `BA.fromByteString` within the `go_fast(_end)` case?

Reply:
Indeed you are right; I clearly lost the forest from the trees here.